src/v2i-hub/MUSTSensorDriverPlugin/src/MUSTSensorDriverPlugin.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#pragma once
+
+#include <PluginClientClockAware.h>
+#include <UdpServer.h>
+#include <ThreadWorker.h>
+#include <mutex>
+#include <SensorDetectedObject.h>
+
+#include "MUSTSensorDetection.h"
+
+
+
+
+namespace MUSTSensorDriverPlugin
+{
+    /**
+     * @brief TODO Plugin description
+     */
+    class MUSTSensorDriverPlugin : public tmx::utils::PluginClientClockAware
+    {
+        private:
+            std::mutex _configMutex;
+            /**
+             * @brief Status label simulation time to be displayed by each plugin.
+             */
+            const char* keyMUSTSensorConnectionStatus = "MUST Sensor Connection Status";
+
+            std::unique_ptr<tmx::utils::UdpServer> mustSensorPacketReceiver;
+
+            std::unique_ptr<tmx::utils::ThreadTimer> mustSensorPacketReceiverThread;
+
+            std::string sensorId;
+
+            std::string projString;
+
+            bool connected = false;
+
+            // Message receiver thread id
+            int mustSensorPacketReceiverThreadId;
```

Comment:
What is the purpose of having this id? It does not seems to be used.

Reply:
It is currently unused but if we ever needed to access this thread, this ID would be required.